src/deploy/functions/runtimes/golang/index.ts

```diff
@@ -56,113 +63,119 @@ export async function tryCreateDelegate(
   return new Delegate(projectId, sourceDir, runtime, module);
 }
 
-// A module can be much more complicated than this, but this is all we need so far.
-// For a full reference, see https://golang.org/doc/modules/gomod-ref
-interface Module {
-  module: string;
-  version: string;
-  dependencies: Record<string, string>;
-}
-
-export function parseModule(mod: string): Module {
-  const module: Module = {
-    module: "",
-    version: "",
-    dependencies: {},
-  };
-  const lines = mod.split("\n");
-  let inRequire = false;
-  for (const line of lines) {
-    if (inRequire) {
-      const endRequireMatch = /\)/.exec(line);
-      if (endRequireMatch) {
-        inRequire = false;
-        continue;
-      }
-
-      const requireMatch = /([^ ]+) (.*)/.exec(line);
-      if (requireMatch) {
-        module.dependencies[requireMatch[1]] = requireMatch[2];
-        continue;
-      }
-
-      if (line.trim()) {
-        logger.debug("Don't know how to handle line", line, "inside a mod.go require block");
-      }
-      continue;
-    }
-    const modMatch = /^module (.*)$/.exec(line);
-    if (modMatch) {
-      module.module = modMatch[1];
-      continue;
-    }
-    const versionMatch = /^go (\d+\.\d+)$/.exec(line);
-    if (versionMatch) {
-      module.version = versionMatch[1];
-      continue;
-    }
-
-    const requireMatch = /^require ([^ ]+) (.*)$/.exec(line);
-    if (requireMatch) {
-      module.dependencies[requireMatch[1]] = requireMatch[2];
-      continue;
-    }
-
-    const requireBlockMatch = /^require +\(/.exec(line);
-    if (requireBlockMatch) {
-      inRequire = true;
-      continue;
-    }
-
-    if (line.trim()) {
-      logger.debug("Don't know how to handle line", line, "in mod.go");
-    }
-  }
-
-  if (!module.module) {
-    throw new FirebaseError("Module has no name");
-  }
-  if (!module.version) {
-    throw new FirebaseError(`Module ${module.module} has no go version`);
-  }
-
-  return module;
-}
-
 export class Delegate {
   public readonly name = "golang";
 
   constructor(
     private readonly projectId: string,
     private readonly sourceDir: string,
     public readonly runtime: runtimes.Runtime,
-    private readonly module: Module
+    private readonly module: gomod.Module
   ) {}
   validate(): Promise<void> {
-    throw new FirebaseError("Cannot yet analyze Go source code");
+    return Promise.resolve();
   }
 
-  build(): Promise<void> {
-    const res = spawn.sync("go", ["build"], {
+  async build(): Promise<void> {
+    try {
+      await promisify(fs.mkdir)(path.join(this.sourceDir, "autogen"));
+    } catch (err) {
+      if (!/EEXIST/.exec(err?.message)) {
```

Comment:
Again if `err?.code` is available that feels better, and if not I think I prefer `err?.message?.includes("EEXIST")`